src/metabase/util/ssh.clj

```diff
@@ -1,32 +1,63 @@
 (ns metabase.util.ssh
-  (:require [clojure.string :as str]
-            [clojure.tools.logging :as log]
+  (:require [clojure.tools.logging :as log]
             [metabase.util :as u])
-  (:import com.jcraft.jsch.JSch))
+  (:import java.io.ByteArrayInputStream
+           [org.apache.sshd.client.future AuthFuture ConnectFuture]
+           org.apache.sshd.client.session.ClientSession
+           org.apache.sshd.client.session.forward.PortForwardingTracker
+           org.apache.sshd.client.SshClient
+           [org.apache.sshd.common.config.keys FilePasswordProvider FilePasswordProvider$ResourceDecodeResult]
+           org.apache.sshd.common.session.SessionHolder
+           org.apache.sshd.common.util.GenericUtils
+           org.apache.sshd.common.util.io.resource.AbstractIoResource
+           org.apache.sshd.common.util.net.SshdSocketAddress
+           org.apache.sshd.common.util.security.SecurityUtils
+           org.apache.sshd.server.forward.AcceptAllForwardingFilter))
 
 (def ^:private default-ssh-timeout 30000)
 
+(def ^:private client
```

Comment:
1.  You could just tag this `^SshClient client` so you don't have to tag it every time you use it

2.  Do you have a sense of whether the startup time/memory overhead of starting the client is minimal enough that it makes sense to start the client at launch regardless of whether it's being used (as opposed to making this a delay and only starting the client if we are creating an SSH tunnel)?

Reply:
1. okay, will do
2. Not sure on that. I went with this approach since you're only meant to have one `SshClient` instance per JVM and atoms seemed overkill / a performance burden.

Reply:
You could do something like this 

```clj
(def ^:private client
  (delay
    (doto (SshClient/setUpDefaultClient)
      (.start)
      (.setForwardingFilter AcceptAllForwardingFilter/INSTANCE))))
```

This still guarantees you only have one `SshClient`, but initializes it the first time you do `@client`.

However you'd still have to use tags `@client`. You could instead change it to a function that derefs the delay e.g.

```clj
(def ^:private ^{:arglists '([])} ^SshClient client
  (partial deref (delay
                   (doto (SshClient/setUpDefaultClient)
                     (.start)
                     (.setForwardingFilter AcceptAllForwardingFilter/INSTANCE)))))
```

and then use `(client)` everywhere and it will be tagged correctly